icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -286,14 +419,34 @@ private static LongNameHandler forCompoundUnit(
             // Some "one" pattern may not contain "{0}". For example in "ar" or "ne" locale.
             String secondaryCompiled = SimpleFormatterImpl
                     .compileToStringMinMaxArguments(secondaryFormat, sb, 0, 1);
-            String secondaryString = SimpleFormatterImpl.getTextWithNoArguments(secondaryCompiled)
-                    .trim();
+            String secondaryFormatString = SimpleFormatterImpl.getTextWithNoArguments(secondaryCompiled);
+
+            // TODO: do not use regular expression
+            String secondaryString = secondaryFormatString.replaceAll("(^\\h*)|(\\h*$)",""); // Trim all spaces.
```

Comment:
I'm okay with this for now. Thanks for the TODO.

Reply:
+1